docs/using-the-compiler.rst

```diff
@@ -0,0 +1,175 @@
+        "abc": "contract b{}", // specify source directly
+        // (axic) I think 'keccak' on its on is not enough. I would go perhaps with swarm: "0x12.." and ipfs: "Qma..." for simplicity
+        // (chriseth) Where the content is stored is a second component, but yes, we could give an indication there.
+        "def": {keccak: "0x123..."}, // source has to be retrieved by its hash
+        "ghi": {file: "/tmp/path/to/file.sol"}, // file on filesystem
+        // (axic) I'm inclined to think the source _must_ be provided in the JSON,
+
+        "dir/file.sol": "contract a {}"
+      },
+      settings:
+      {
+        remappings: [":g/dir"], // just as it used to be
+        // (axic) what is remapping doing exactly?
+        optimizer: {enabled: true, runs: 500},
+        // if given, only compiles this contract, can also be an array. If only a contract name is given, tries to find it if unique.
+        compilationTarget: "myFile.sol:MyContract",
+        // addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different
+        libraries: {
+          "def:MyLib": "0x123123..."
+        },
+        // The following can be used to restrict the fields the compiler will output.
+        // (axic)
+        outputSelection: [
+            "abi", "evm.assembly", "evm.bytecode", ..., "why3", "ewasm.wasm"
+        ]
+        outputSelection: {
+        abi,asm,ast,bin,bin-runtime,clone-bin,devdoc,interface,opcodes,srcmap,srcmap-runtime,userdoc
+
+ --ast                 AST of all source files.
+  --ast-json            AST of all source files in JSON format.
+  --asm                 EVM assembly of the contracts.
+  --asm-json            EVM assembly of the contracts in JSON format.
+  --opcodes             Opcodes of the contracts.
+  --bin                 Binary of the contracts in hex.
+  --bin-runtime         Binary of the runtime part of the contracts in hex.
+  --clone-bin           Binary of the clone contracts in hex.
+  --abi                 ABI specification of the contracts.
+  --interface           Solidity interface of the contracts.
+  --hashes              Function signature hashes of the contracts.
+  --userdoc             Natspec user documentation of all contracts.
+  --devdoc              Natspec developer documentation of all contracts.
+  --formal              Translated source suitable for formal analysis.
+
+          // to be defined
+        }
+      }
+    }
+
+
+Regular Output
+--------------
+
+.. code-block:: none
+
+    {
+      errors: ["error1", "error2"], // we might structure them
+      errors: [
+          {
+              // (axic)
+              file: "sourceFile.sol", // optional?
+              contract: "contractName", // optional
+              line: 100, // optional - currently, we always have a byte range in the source file
```

Comment:
`sourceLocation`:  { "start": 0, "end": 100 }
`secondarySourceLocation`: { "start": 10, "end": 50 }

Byte ranges in the source file.